editor/src/core/model/element-template-utils.spec.tsx

```diff
@@ -739,3 +749,245 @@ describe('rearrangeJsxChildren', () => {
+    const foundElement = findJSXElementChildAtPath(
+      getComponentsFromTopLevelElements(file.topLevelElements),
+      path,
+    )
+    return foundElement
+  }
+
+  function expectElementAtPathHasMatchingUID(file: ParseSuccess, pathString: string) {
+    const foundElement = findElement(file, pathString)
+
+    expect(foundElement).toBeDefined()
+    expect(getUtopiaID(foundElement!)).toEqual(toUid(fromStringStatic(pathString)))
+  }
+
+  function expectElementAtPathHasMatchingUIDForPaths(file: ParseSuccess, paths: Array<string>) {
+    paths.forEach((path) => expectElementAtPathHasMatchingUID(file, path))
+  }
+
+  function expectElementFoundNull(file: ParseSuccess, paths: Array<string>) {
+    paths.forEach((path) => expect(findElement(file, path)).toBeNull())
+  }
+
+  it('simple project with divs', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            <div data-uid='child-d' />
+            <div data-uid='child-c'>
+              <div data-uid='grandchild-c' />
+            </div>
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/child-a',
+    ])
+    expectElementFoundNull(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/not-existing-child',
+    ])
+  })
+
+  it('simple project with divs with a conditional expression as sibling', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            {true ? 
+              (
+                <div data-uid='ternary-true-root'>
+                  <div data-uid='ternary-true-child' />
+                </div> 
+              ) : (
+                <div data-uid='ternary-false-root'>
+                  <div data-uid='ternary-false-child' />
+                </div> 
+              )
+            }
+            <div data-uid='child-d' />
+            <div data-uid='child-c'>
+              <div data-uid='grandchild-c' />
+            </div>
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/child-a',
+    ])
+
+    // !!!! This is a deliberately failing test to demonstrate the issue Berci found
```

Comment:
I wanted to commit the breaking test first, but I have a one-liner fix to this problem.